sequencer/src/l1_client.rs

```diff
@@ -88,167 +85,62 @@ impl Committable for L1BlockInfo {
-                // maintaining our invariants (increasing L1 block numbers) over UX.
-                tracing::warn!(
-                    "got L1 head {head} which is not newer than previous head {}",
-                    snapshot.head
-                );
-                continue;
-            }
-
-            // A new block has been produced. This happens fairly rarely, so it is now ok to poll to
-            // see if a new block has been finalized.
-            let finalized = match get_finalized_block(&rpc, finalized_tag).await {
-                Ok(finalized) => Some(finalized),
-                Err(err) => {
-                    tracing::error!("failed to get finalized block: {err}");
-
-                    // If we cannot get the finalized block for any reason, don't let this stop us
-                    // from updating the L1 head. By returning `None` here, we will proceed to
-                    // update the head but not the finalized block, and then we will reset the
-                    // connection, in the hopes that this fixes whatever went wrong with
-                    // `get_finalized_block`.
-                    None
+    /// Proxy to `Provider.get_block_number`.
+    async fn get_block_number(&self) -> u64 {
+        let mut count = 0;
+        loop {
+            match self.provider.get_block_number().await {
+                Ok(n) => return n.as_u64(),
+                Err(e) => {
+                    tracing::warn!("Blocknumber error: {}", e);
+                    sleep(self.retry_delay).await;
                 }
-            };
-
-            // Update the snapshot.
-            let mut snapshot = RwLockUpgradableReadGuard::upgrade(snapshot).await;
-            snapshot.head = head;
-            if let Some(finalized) = finalized {
-                snapshot.finalized = finalized;
             }
-
-            // Drop our exclusive lock before logging anything.
-            let snapshot = RwLockWriteGuard::downgrade(snapshot);
-            tracing::info!("updated L1 snapshot to {:?}", *snapshot);
-
-            // If we encountered an error in `get_finalized_block`, reset the connection now that we
-            // have updated what we can.
-            if finalized.is_none() {
-                tracing::warn!("resetting connection due to error in get_finalized_block");
-                break;
+            assert!(count < 6);
+            count += 1;
+        }
+    }
+    /// Proxy to `get_finalized_block`.
+    async fn get_finalized_block(&self) -> Option<L1BlockInfo> {
+        let mut count = 0;
+        loop {
+            match get_finalized_block(&self.provider).await {
+                Ok(block) => return block,
+                Err(e) => {
+                    tracing::warn!("Finalized block error: {}", e);
+                    sleep(self.retry_delay).await;
+                }
             }
+            assert!(count < 6);
+            count += 1;
```

Comment:
?

Reply:
Its a retry limit so tests fail instead of retrying indefinably. It will be removed once I figure out what is causing tests to fail. Strangely, causing these to panic (instead of retrying indefinably) causes tests to pass. I can see the panics in the test output, but they are being ignored for some reason and since the tests no longer fail on timeout, they are marked as successful.

Reply:
removed in edb82e76b40af46867714115d16373285d074c8b